app/scripts/com/2fdevs/videogular/controllers/vg-controller.js

```diff
@@ -362,12 +374,39 @@ angular.module("com.2fdevs.videogular")
         };
 
         this.play = function () {
-            this.mediaElement[0].play();
+            var mediaElement = this.mediaElement[0];
+
+            // short-circuit if already playing
+            if (playPromise || !mediaElement.paused) {
+                return;
+            }
+
+            playPromise = mediaElement.play();
+
+            if (playPromise && playPromise.then && playPromise.catch) { // browser has async play promise
+                playPromise
+                    .then(function() {
+                        playPromise = null;
+                    })
+                    .catch(function() {
+                        // deliberately empty for the sake of eating console noise
+                    });
+            }
+
             this.setState(VG_STATES.PLAY);
         };
 
         this.pause = function () {
-            this.mediaElement[0].pause();
+            var mediaElement = this.mediaElement[0];
+            
+            if (playPromise) { // browser has async play promise
+                playPromise.then(function() {
```

Comment:
When you pause the playPromise is executed? I understand that this is to clear the promise? If that's the case then no problem